protocol/rules/chainedhotstuff.go

```diff
@@ -70,7 +70,11 @@ func (hs *ChainedHotStuff) CommitRule(block *hotstuff.Block) *hotstuff.Block {
 		return nil
 	}
 
-	if block1.Parent() == block2.Hash() && block2.Parent() == block3.Hash() {
+	// since our implementation does not create dummy blocks every view, 
+	// we explicitly check that the parents are in the previous view
```

Comment:
Do you mean both parent and grandparent here? The check below only checks the view of block1 against block3. But perhaps that is enough since the block hashes matches. If so, we should add a comment about that.